operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -233,8 +238,27 @@ func extractGCSConfigSecret(s *corev1.Secret) (*storage.GCSStorageConfig, error)
 		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyGCPServiceAccountKeyFilename)
 	}
 
+	credentialsFile := struct {
+		CredentialsType string `json:"type"`
+	}{}
+
+	err := json.Unmarshal(keyJSON, &credentialsFile)
+	if err != nil {
+		return nil, errGCPParsingCredentialsFile
+	}
+
+	var (
+		audience           = string(s.Data[storage.KeyGCPWorkloadIdentityProviderAudience])
+		isWorkloadIdentity = credentialsFile.CredentialsType == gcpExternalAccountType
+	)
+	if isWorkloadIdentity && len(audience) == 0 {
+		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyGCPWorkloadIdentityProviderAudience)
```

Comment:
Can we provide a "default audience" for GCP like we do for the other integrations?

--edit: Hm... It looks as if GCP does not have a "universal default audience", instead the default is the "full canonical resource name" of the authentication provider. Ok, so the user needs to provide _something_. :shrug: